content/general/concepts/what-is-the-factory-pattern/what-is-the-factory-pattern.md

```diff
@@ -0,0 +1,265 @@
+---
+Title: 'Factory Pattern'
+Description: 'The factory pattern defers instantiation logic of a parent abstract class to its concrete sub-classes.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The factory pattern defers instantiation logic of a parent abstract class to its concrete sub-classes. At time of object creation, the specific class type may not be known, in which a creator class' factory method is used to decouple this identifying logic. This is typically done through the methods, parameters, and a switch statement.
+
+## UML Design
+
+![UML diagram of a factory](https://raw.githubusercontent.com/Codecademy/docs/main/media/factory-uml.png)
+
+## Example
+
+To illustrate the factory pattern, below provides a real-world example, in Java, depicting potential considerations for a taxi and food delivery booking application. A user of the app will need to send a request for either a standard taxi, a large taxi, a mini bus, or wish to have food delivered from a restaurant. However, before the program understands their needs, the needed request needed remains unknown. When a request object is created, an estimated price is calculated and returned.
+
+Below provides a parent abstract class of a taxi request:
+
+```java
+public abstract class TaxiRequest<T> {
+  protected T pickUpLocation;
+  protected String destination;
+  protected double distance;
+  protected double estimatedPrice;
+
+  public TaxiRequest(final T pickUpLocation, final String destination, final int distance) {
+    this.createTaxiRequest(pickUpLocation, destination, distance);
+  }
+
+  public abstract void createTaxiRequest(final T pickUpLocation, final String destination, final int distance);
+
+
+  // Getters and setters
```

Comment:
Some dots or ellipsis here to indicate omitted code.

Reply:
The only code that is missing here is the getters and setters indicated by the comment. The createTaxiRequest() is stubbed out as its an abstract method. This is just to enforce the sub classes to have a method with the same signature. 

I have resolved, but if I mis understood, please feel free to re-open